generator/osm_source.cpp

```diff
@@ -524,6 +532,30 @@ class MainFeaturesEmitter : public EmitterBase
 };
 }  // anonymous namespace
 
+void SyncOfstream::Open(string const & fullPath)
+{
+  lock_guard<mutex> gard(m_mutex);
+  m_stream.open(fullPath, std::ofstream::out);
+}
+
+bool SyncOfstream::IsOpened()
+{
+  lock_guard<mutex> gard(m_mutex);
+  return m_stream.is_open() && !m_stream.fail();
+}
+
+void SyncOfstream::Write(uint32_t featureId, vector<osm::Id> const & osmIds)
+{
+  if (!IsOpened())
+    return;
+
+  lock_guard<mutex> gard(m_mutex);
+  m_stream << featureId << ", ";
+  for (osm::Id const & osmId : osmIds)
+    m_stream << osmId.OsmId() << ", ";
```

Comment:
So what? A list of items separated by commas is more natural than list of items with trailing comma.